### ISSUE TITLE
Loading model in another session gives low accuracy

### ISSUE THREAD

**Issue by SanjayDevTech** [keras-team/keras#17526]
**System information**.
<details>
<summary>Click to open</summary>

```
== check python ===================================================
python version: 3.8.10
python branch: 
python build version: ('default', 'Nov 14 2022 12:59:47')
python compiler version: GCC 9.4.0
python implementation: CPython


== check os platform ===============================================

== are we in docker =============================================
No

== compiler =====================================================
c++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


== check pips ===================================================
numpy                         1.21.6
proto-plus                    1.22.2
protobuf                      3.19.6
tensorflow                    2.9.2
tensorflow-datasets           4.8.2
tensorflow-estimator          2.9.0
tensorflow-gcs-config         2.9.1
tensorflow-hub                0.12.0
tensorflow-io-gcs-filesystem  0.30.0
tensorflow-metadata           1.12.0
tensorflow-probability        0.17.0

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.version.VERSION = 2.9.2
tf.version.GIT_VERSION = v2.9.1-132-g18960c44ad3
tf.version.COMPILER_VERSION = 9.3.1 20200408
== env ==========================================================
LD_LIBRARY_PATH /usr/lib64-nvidia
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
Sat Feb  4 07:43:33 2023       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.47.03    Driver Version: 510.47.03    CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   54C    P0    26W /  70W |      0MiB / 15360MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

== cuda libs  ===================================================
/usr/local/lib/python3.8/dist-packages/torch/lib/libcudart-45da57e3.so.11.0
/usr/local/lib/python3.8/dist-packages/torchvision.libs/libcudart.a44f4c9b.so.11.0
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudart.so.11.2.152
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudart_static.a

== tensorflow installed from info ==================
Name: tensorflow
Version: 2.9.2
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: /usr/local/lib/python3.8/dist-packages
Required-by: kapre

== python version  ==============================================
(major, minor, micro, releaselevel, serial)
(3, 8, 10, 'final', 0)
```

got this from https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh
</details>

**Describe the problem**.
load_model causes loss of accuracy and increasing in loss across python sessions in Colab
loss - Sparse categorical cross entropy
optimizer - adam

**Describe the current behavior**.
loaded model shows expected accuracy in same session where the model is saved.
loaded model showing low accuracy in another session when I zipped from prev session to this session.

**Describe the expected behavior**.
It should be showing the same accuracy in all sessions.
I have used the same dataset.

**Standalone code to reproduce the issue**.
1. Evaluating in session1:
```python
model.evaluate(x_test, y_test)
```
```bash
533/533 [==============================] - 123s 226ms/step - loss: 0.2000 - accuracy: 0.9352
[0.1999737024307251, 0.9352205395698547]
```
2. Saving model in session 1:
```python
model.save("saved_model")
```
3. Loading again in session 1:
```python
del model
import keras
model = keras.models.load_model("saved_model")
```
4. Evaluating again in session 1:
```python
model.evaluate(x_test, y_test)
```
```bash
533/533 [==============================] - 125s 234ms/step - loss: 0.2000 - accuracy: 0.9352
[0.1999737024307251, 0.9352205395698547]
```
Since it was the same data, it shows same accuracy.

5. Loading in session 2:
```python
model = keras.models.load_model("saved_model")
```
6. Evaluating
```python
model.evaluate(x_test, y_test)
```
```bash
533/533 [==============================] - 8s 8ms/step - loss: 0.5370 - accuracy: 0.8465
```

**Comment by SanjayDevTech** [comment]
@sushreebarsa I think this is same as https://github.com/keras-team/keras/issues/17511

**Comment by sushreebarsa** [comment]
@SanjayDevTech Thank you for the update!
Could you share the complete standalone code to replicate the issue?
Thank you!
